File: frontend/src/utils/statusTranslator.ts
export const StatusTranslator = {
  orderStatusMap: new Map<string, string>([
    ['PENDING', '待处理'],
    ['CONFIRMED', '已确认'],
    ['SHIPPED', '已发货'],
    ['RECEIVED', '已收货'],
    ['CANCELLED', '已取消'],
    ['COMPLETED', '已完成'],
    ['PENDING_SETTLEMENT', '待结算'],
    ['PARTIAL_SETTLED', '部分结算'],
    ['SETTLED', '已结算'],
  ]),

  shippingStatusMap: new Map<string, string>([
    ['PENDING', '待发货'],
    ['SHIPPED', '已发货'],
    ['IN_TRANSIT', '运输中'],
    ['DELIVERED', '已送达'],
    ['RECEIVED', '已收货'],
    ['RETURNED', '已退货'],
  ]),

  inboundStatusMap: new Map<string, string>([
    ['PENDING', '待处理'],
    ['SHIPPED', '已发货'],
    ['RECEIVED', '已收货'],
    ['CANCELLED', '已取消'],
  ]),

  translateStatus(statusName: string | null | undefined): string {
    if (!statusName || statusName === 'NULL' || statusName === 'None') return '无';
    
    let translated = this.orderStatusMap.get(statusName);
    if (translated) return translated;
    
    translated = this.shippingStatusMap.get(statusName);
    if (translated) return translated;
    
    translated = this.inboundStatusMap.get(statusName);
    if (translated) return translated;
    
    return statusName;
  },

  translateOrderStatus(status: string | null | undefined): string {
    if (!status || status === 'NULL') return '未知';
    return this.orderStatusMap.get(status) || status;
  },

  translateShippingStatus(status: string | null | undefined): string {
    if (!status || status === 'NULL') return '未知';
    return this.shippingStatusMap.get(status) || status;
  },

  translateInboundStatus(status: string | null | undefined): string {
    if (!status || status === 'NULL') return '未知';
    return this.inboundStatusMap.get(status) || status;
  },
};

export default StatusTranslator;

File: backend/src/main/java/com/supplypro/utils/StatusTranslator.java
package com.supplypro.utils;

import com.supplypro.entity.PurchaseOrder;
import com.supplypro.entity.InboundOrder;
import java.util.HashMap;
import java.util.Map;

public class StatusTranslator {
    
    private static final Map<String, String> ORDER_STATUS_MAP = new HashMap<>();
    private static final Map<String, String> SHIPPING_STATUS_MAP = new HashMap<>();
    private static final Map<String, String> INBOUND_STATUS_MAP = new HashMap<>();
    private static final Map<String, String> DELIVERY_METHOD_MAP = new HashMap<>();
    
    static {
        ORDER_STATUS_MAP.put("PENDING", "待处理");
        ORDER_STATUS_MAP.put("CONFIRMED", "待发货");
        ORDER_STATUS_MAP.put("SHIPPED", "已发货");
        ORDER_STATUS_MAP.put("RECEIVED", "已收货");
        ORDER_STATUS_MAP.put("CANCELLED", "已取消");
        ORDER_STATUS_MAP.put("COMPLETED", "已完成");
        ORDER_STATUS_MAP.put("PENDING_SETTLEMENT", "待结算");
        ORDER_STATUS_MAP.put("PARTIAL_SETTLED", "部分结算");
        ORDER_STATUS_MAP.put("SETTLED", "已结算");
        
        SHIPPING_STATUS_MAP.put("PENDING", "待发货");
        SHIPPING_STATUS_MAP.put("SHIPPED", "已发货");
        SHIPPING_STATUS_MAP.put("IN_TRANSIT", "运输中");
        SHIPPING_STATUS_MAP.put("DELIVERED", "已送达");
        SHIPPING_STATUS_MAP.put("RECEIVED", "已收货");
        SHIPPING_STATUS_MAP.put("RETURNED", "已退货");
        
        INBOUND_STATUS_MAP.put("PENDING", "待处理");
        INBOUND_STATUS_MAP.put("SHIPPED", "已发货");
        INBOUND_STATUS_MAP.put("RECEIVED", "已收货");
        INBOUND_STATUS_MAP.put("CANCELLED", "已取消");
        
        // 配送类型中文映射
        DELIVERY_METHOD_MAP.put("SelfDelivery", "自配送");
        DELIVERY_METHOD_MAP.put("Logistics", "物流配送");
        DELIVERY_METHOD_MAP.put("Express", "快递配送");
        DELIVERY_METHOD_MAP.put("Self", "自配送");
    }
    
    public static String translateOrderStatus(PurchaseOrder.Status status) {
        if (status == null) return "未知";
        return ORDER_STATUS_MAP.getOrDefault(status.name(), status.name());
    }
    
    public static String translateOrderStatus(String statusName) {
        if (statusName == null || "NULL".equals(statusName)) return "无";
        return ORDER_STATUS_MAP.getOrDefault(statusName, statusName);
    }
    
    public static String translateShippingStatus(PurchaseOrder.ShippingStatus status) {
        if (status == null) return "未知";
        return SHIPPING_STATUS_MAP.getOrDefault(status.name(), status.name());
    }
    
    public static String translateShippingStatus(String statusName) {
        if (statusName == null || "NULL".equals(statusName)) return "无";
        return SHIPPING_STATUS_MAP.getOrDefault(statusName, statusName);
    }
    
    public static String translateInboundStatus(InboundOrder.Status status) {
        if (status == null) return "未知";
        return INBOUND_STATUS_MAP.getOrDefault(status.name(), status.name());
    }
    
    public static String translateInboundStatus(String statusName) {
        if (statusName == null || "NULL".equals(statusName)) return "无";
        return INBOUND_STATUS_MAP.getOrDefault(statusName, statusName);
    }
    
    public static String translateStatus(String statusName) {
        if (statusName == null || "NULL".equals(statusName)) return "无";
        String translated = ORDER_STATUS_MAP.get(statusName);
        if (translated != null) return translated;
        translated = SHIPPING_STATUS_MAP.get(statusName);
        if (translated != null) return translated;
        translated = INBOUND_STATUS_MAP.get(statusName);
        if (translated != null) return translated;
        return statusName;
    }
    
    public static String translateDeliveryMethod(String deliveryMethod) {
        if (deliveryMethod == null || "NULL".equals(deliveryMethod)) return "未知";
        return DELIVERY_METHOD_MAP.getOrDefault(deliveryMethod, deliveryMethod);
    }
    
    private StatusTranslator() {}
}
